app/src/main/java/org/glucosio/android/db/DatabaseHandler.java

```diff
@@ -931,45 +924,42 @@ public void deleteCholesterolReading(CholesterolReading reading) {
         return idArray;
     }
 
-    public ArrayList<Integer> getHDLCholesterolReadingAsArray() {
+    public List<Double> getHDLCholesterolReadingAsArray() {
         List<CholesterolReading> readings = getCholesterolReadings();
-        ArrayList<Integer> readingArray = new ArrayList<Integer>();
+        ArrayList<Double> readingArray = new ArrayList<>(readings.size());
         int i;
 
         for (i = 0; i < readings.size(); i++) {
-            int reading;
             CholesterolReading singleReading = readings.get(i);
-            reading = singleReading.getHDLReading();
+            double reading = singleReading.getHDLReading();
             readingArray.add(reading);
         }
 
         return readingArray;
     }
 
-    public ArrayList<Integer> getLDLCholesterolReadingAsArray() {
+    public List<Double> getLDLCholesterolReadingAsArray() {
         List<CholesterolReading> readings = getCholesterolReadings();
-        ArrayList<Integer> readingArray = new ArrayList<Integer>();
+        ArrayList<Double> readingArray = new ArrayList<>(readings.size());
         int i;
 
         for (i = 0; i < readings.size(); i++) {
-            int reading;
             CholesterolReading singleReading = readings.get(i);
-            reading = singleReading.getLDLReading();
+            double reading = singleReading.getLDLReading();
             readingArray.add(reading);
         }
 
         return readingArray;
     }
 
-    public ArrayList<Integer> getTotalCholesterolReadingAsArray() {
+    public List<Double> getTotalCholesterolReadingAsArray() {
```

Comment:
The method name does not reflect the return type.

Reply:
Also renamed